### DISCUSSION TITLE
Looking for example workflow

### DISCUSSION THREAD

**Discussion by dakoellis** [PaulHatch/semantic-version#132] [Q&A]
Is there one somewhere?  I was expecting to be able to use the version_format to create a tag with an increment, but it seems the tag must be in 1.2.3 format (or with a prefix).  

My goal is to be able to tag something with `version_format: "${major}.${minor}.${patch}-${increment)"` and have that go through to be able to 

1, avoid force pushing tags, and 
2, be able to track a version through multiple envs and not just at release time.  

Is this possible using this action? did I miss some option somewhere?

**Comment by PaulHatch** [comment]
Hey @dakoellis ,
There is indeed a [guide](https://github.com/PaulHatch/semantic-version/blob/master/guide.md) which can help you select a strategy and includes examples of the necessary configuration for each.

Regarding your `version_format` is not the tag, that's just a convenience for formatting the output however you like. The tag must indeed be in the format `{prefix}{major}.{minor}.{patch}{optional suffix}`. The version tag goes through a multi-stage parsing process, first using glob patterns to pull possible tag candidates using Git, then filtering the results using regular expression. Some values must also be escaped to work with globs, so providing a similar single format string is not really practical, instead you can specify optional prefix/suffix. The action provides and output indicating what the tag for the current version should be.

Increments are always derived from the repo commit history. There is no need to include them in a tag, though there is also nothing stopping you from adding the increment to a tag, which will just be ignored by the action.
